### PR TITLE
chore: Add new SDK overhead page for Android

### PR DESCRIPTION
**Pull request by kahest**
## DESCRIBE YOUR PR

Adds new "SDK Overhead" menu item + page below the "Mobile SDK Releases" item. Covers Errors etc. in more detail, links to existing product pages for Performance+Profiling.

Once https://github.com/getsentry/sentry-docs/pull/11989 is merged we'll add a section on SR and link to the appropriate page (but we'll not block this PR for it).

fixes https://github.com/getsentry/projects/issues/452

Next steps/future work: see https://github.com/getsentry/projects/issues/169

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [x] Other deadline: no specific date, target is EoW
- [ ] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [x] Checked Vercel preview for correctness, including links
- [x] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [x] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

**Comment by romtsn** [comment]
heads up: https://github.com/getsentry/sentry-docs/pull/11989 has been merged!

File: docs/platforms/android/overhead/index.mdx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+title: SDK Overhead
+description: "Learn about Sentry's Android SDK overhead."
+sidebar_order: 7500
+---
+
+Adding new features or dependencies to your app incurs additional costs on resources - CPU time, memory usage, network bandwidth, among other things. Sentry SDKs are no different. This document adds transparency to the possible overhead that using this SDK can add, and help you find the feature set and configurations that work best for you.
+
+## General Approach
+
+The SDK is designed to have minimal to no impact on the user experience. To achieve this, we utilize an array of tools to measure and optimize the performance of our implementations continuously.
```
Comment (coolguyzone) [inline docs/platforms/android/overhead/index.mdx:11]:
```suggestion
Our SDK is designed to have minimal to no impact on user experience. To achieve this, we utilize an array of tools to continuously measure and optimize the performance of our implementations.
```

File: docs/platforms/android/overhead/index.mdx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+If you find (e.g. via local Profiling, or using Sentry to improve the performance of your app) that the SDK does not operate within the guidelines mentioned below, please [open an issue](https://github.com/getsentry/sentry-java/issues/new/choose) on our SDK repo and make sure you provide as much context as you can.
+
+## Error Monitoring
+
+During regular operation, error monitoring incurs little to no overhead. Once an error or crash occurs, the user experience is compromised, and any crash handling routines operate under time constraints imposed by the system. This means that these implementations are highly optimized to perform the required work as quickly as possible.
+
+The SDK also provides methods to manually capture events: <PlatformLink to="/usage/#capturing-errors">`captureError`</PlatformLink> and <PlatformLink to="/usage/#capturing-messages">`captureMessage`</PlatformLink>. These methods perform some complex operations, such as capturing stack trace information, and while they are highly optimized as well, calling them in tight loops should be avoided.
```
Comment (coolguyzone) [inline docs/platforms/android/overhead/index.mdx:21]:
```suggestion
Our SDK also provides methods to manually capture events: <PlatformLink to="/usage/#capturing-errors">`captureError`</PlatformLink> and <PlatformLink to="/usage/#capturing-messages">`captureMessage`</PlatformLink>. These methods perform some complex operations, such as capturing stack trace information, and while they are highly optimized as well, calling them in tight loops should be avoided.
```

File: docs/platforms/android/overhead/index.mdx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+If your app raises many errors in a tight loop, it can become too much to process quickly enough, and UI jank can be the result, so make sure you handle such cases appropriately. The SDK provides callbacks to fine-tune when to capture <PlatformLink to="/enriching-events/screenshots#customize-screenshot-capturing">Screenshots</PlatformLink> and <PlatformLink to="/enriching-events/viewhierarchy#customize-view-hierarchy-capturing">View Hierarchy</PlatformLink>, which can also be used to reduce performance impact in scenarios where you don't need screenshots or view hierarchies.
+
+## Breadcrumbs
+
+<PlatformLink to="/enriching-events/breadcrumbs">Breadcrumbs</PlatformLink> are collected through automated integrations or by manually adding them. To have them readily available for every event generated by the SDK, they are continuously persisted, and managed in a performant buffer. This is expected to have no impact on user experience.
```
Comment (coolguyzone) [inline docs/platforms/android/overhead/index.mdx:31]:
```suggestion
<PlatformLink to="/enriching-events/breadcrumbs">Breadcrumbs</PlatformLink> are collected through automated integrations or by manually adding them. To have them readily available for every event generated by the SDK, they are continuously persisted, and managed in a performant buffer. This shouldn't impact user experience.
```

File: docs/platforms/android/overhead/index.mdx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+Capturing excessive numbers of breadcrumbs (e.g. by creating breadcrumbs for all log messages) can cause significant performance overhead. To mitigate this, review and adapt your app's usage of breadcrumbs. For example, increase the min-level of log messages for which to create breadcrumbs from `warn` to `error`.
```
Comment (coolguyzone) [inline docs/platforms/android/overhead/index.mdx:33]:
```suggestion
Capturing excessive numbers of breadcrumbs (for example, creating breadcrumbs for all log messages) can cause significant performance overhead. To mitigate this, review and adapt your app's usage of breadcrumbs. For example, increase the min-level of log messages that create breadcrumbs from `warn` to `error`.
```

**Review by coolguyzone — APPROVED** [review]
Thanks for adding this! Threw in a few minor suggestions, looks good to me! 🫡

**Pull request merged**